lib/trebuchet/backend/redis_hammerspaced.rb

```diff
@@ -0,0 +1,102 @@
+require 'trebuchet/backend/redis'
+require 'json'
+
+class Trebuchet::Backend::RedisHammerspaced < Trebuchet::Backend::Redis
+
+  # This class will rely on a cron job to sync all trebuchet features
+  # to local hammerspace thus this class never directly updates hammerspace
+  # We also cache in memory the features and rely on before_filter
+  # to lazily invalidate local cache
+
+  attr_accessor :namespace
+
+  def initialize(*args)
+    @namespace = 'trebuchet/'
+    begin
+      # args.first must be a hash
+      @options = args.first
+      @redis = @options[:client]
+      @hammerspace = @options[:hammerspace]
+      unless @options[:skip_check]
+        # raise error if not connectedUncaught ReferenceError: google is not defined
+        @redis.exists(feature_names_key) # @redis.info is slow and @redis.client.connected? is NOT reliable
+        @hammerspace.has_key?(feature_names_key)
+      end
+    rescue Exception => e
+      raise Trebuchet::BackendInitializationError, e.message
+    end
+  end
+
+  def get_strategy(feature_name)
+    if cached_strategies.has_key?(feature_name)
+      # use cached if available (even if value is nil)
+      cached_strategies[feature_name]
+    else
+      # call to hammerspace
+      cache_strategy feature_name, get_strategy_hammerspace(feature_name)
+    end
+  end
+
+  def get_strategy_hammerspace(feature_name)
+    # Read from hammerspace
+    return nil unless @hammerspace.has_key?(feature_key(feature_name))
+    # h will be a string, we need to convert it back to Hash
+    h = @hammerspace[feature_key(feature_name)]
```

Comment:
Hammerspace only stores string keys, and strings are truthy in ruby, so we could probably avoid an extra string concatenation and an extra hammerspace lookup by doing:

```ruby
h = @hammerspace[feature_key(feature_name)]
return nil unless h
...
```